### PR TITLE
Create standard error module

### PR DESCRIPTION
**Pull request by johnBartos**
### What does this Pull Request do?
Creates an Error module which standardizes errors through numerical code values

### Why is this Pull Request needed?
Errors throughout the player are hand-coded, inconsistent, and hard to program against. Using a standardized numerical system will simplify our code base and ensure errors are uniform across plugins

### Are there any points in the code the reviewer needs to double check?
This module is currently unused in this project but used in Commercial - reviews here are limited to feedback specifically about the structure and approach to this module. Error codes and categories can be decided & changed in a later PR.

### Are there any Pull Requests open in other repos which need to be merged with this?
https://github.com/jwplayer/jwplayer-commercial/pull/3348

#### Addresses Issue(s):

JW7-2606

### CODE REVIEW

**Comment by johnBartos** [comment]
🙄 browserstack timeout

File: src/js/utils/error.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    // Crossdomain error
+    CROSSDOMAIN_ERROR: 4002,
+    // Accessing HTTP over HTTPS
+    BLOCKED_MIXED_CONTENT: 4003
+};
+
+export const errorSeverity = {
+    RECOVERABLE: 1,
+    FATAL: 2
+};
```
Comment (robwalch) [inline src/js/utils/error.js]: Do these need to be maps? Can we export each const so they all become variables that can be minified?

Ex: `export const CATEGORY_PLAYER = 1;`

Comment (johnBartos) [inline src/js/utils/error.js]: I'd rather have a file `error-category.js` and inside `export const PLAYER = 1` such that you can `import * as ErrorCategory from error-category`. Need to balance developer ease with minification.

Comment (johnBartos) [inline src/js/utils/error.js]: Update: It seems that doing it your way does not allow minification either.

**Comment by johnBartos** [comment]
Closed since this is now its own repo

**Pull request closed without merging**